pubsub/system-test/subscriptions.test.js

```diff
@@ -35,39 +36,56 @@ describe('pubsub:subscriptions', function () {
 
   describe('createSubscription', function () {
     it('should create a subscription', function (done) {
-      program.createSubscription(topicName, subscriptionName, function (err, subscription) {
+      program.createSubscription(topicName, subscriptionName, function (err, subscription, apiResponse) {
         assert.ifError(err);
-        assert.equal(subscription.name, name);
+        assert.equal(subscription.name, fullSubscriptionName);
         assert(console.log.calledWith('Created subscription %s to topic %s', subscriptionName, topicName));
-        // The next test sometimes fails, so, slow this test down
+        assert.notEqual(apiResponse, undefined);
+        // Listing is eventually consistent, so give the index time to update
         setTimeout(done, 5000);
       });
     });
   });
 
+  describe('getSubscriptionMetadata', function () {
+    it('should get metadata for a subscription', function (done) {
+      program.getSubscriptionMetadata(subscriptionName, function (err, metadata) {
+        assert.ifError(err);
+        assert.equal(metadata.name, fullSubscriptionName);
+        assert(console.log.calledWith('Got metadata for subscription: %s', subscriptionName));
+        done();
+      });
+    });
+  });
+
   describe('listSubscriptions', function () {
     it('should list subscriptions', function (done) {
       program.listSubscriptions(topicName, function (err, subscriptions) {
```

Comment:
Can/should we `assert` that this is somehow different than `listAllSubscriptions`? e.g. run through all the results and make sure they have the right topicName?